packages/core/src/types.ts

```diff
@@ -56,10 +75,7 @@ export type Event<TEvent extends EventObject> = TEvent['type'] | TEvent;
 export interface ActionMeta<
   TContext,
   TEvent extends EventObject,
-  TAction extends ActionObject<TContext, TEvent> = ActionObject<
```

Comment:
Changes like this across the codebase are actually bug fixes~ since the constraint and the default was incorrect previously.